src/cookies.js

```diff
@@ -247,14 +247,19 @@ function checkOriginForSettingCookie(win, options, name) {
     );
     return;
   }
+  const metaTag =
+    win.document.head &&
+    win.document.head.querySelector("meta[name='runtime-host']");
+
   userAssert(
-    !isProxyOrigin(win.location.href),
+    !isProxyOrigin(win.location.href) || metaTag != null,
```

Comment:
@erwinmombay - after the validator is updated, could you verify pages that are sent to the AMP cache strip out this meta tag? In Jake's earlier commits, this assert ensured a page viewed directly from the AMP cache wouldn't allow a cookie to be set. Since the self hosst exception was added here as well, we need to make sure `<meta name="runtime-host" content="">` won't slip through the caching process and allow cookies to be set on cdn.ampproject.org.

Reply:
will do. thanks

Reply:
circling back on this. we currently dont have anything in place that does the `<meta name="runtime-host" content="">` removal. my understanding is that we need this to get this all working securely. I can start working on this today if we need it and have it deployed by next week but wanted to check with you if my understanding is correct.

Reply:
Thanks for following-up on this. It would be best to strip `<meta name="runtime-host" ...>` tags from AMP pages before they are admitted into the AMP Project's cache. It would be fantastic if this could be done in a way that the pages are not considered invalid, just that this meta tag is removed before caching.

If I just focus on this function, both of the `userAsert()` calls that check whether the current origin is a proxy origin have overrides now. The simple presence of `<meta name="runtime-host" ...>` is enough to allow cookies to be written, regardless of `content=...` value. I think this is an allowable condition iff the AMP Project strips the tag before caching.

Reply:
got it. deployment to our ingestion stack might take a few weeks just as a heads up. im going to submit the internal changes for review